src/dal/zodiacdeep.ts

```diff
@@ -0,0 +1,126 @@
+import { database } from "../libs/database";
+import { SignData } from "../models/signData";
+import { firestore } from "firebase-admin";
+import { Signs_ZodiacDirBG } from "../models/signs";
+import { ISignRecord } from "../interfaces/signs";
+
+export async function GetArrayJSNatFromSignData(
+  sign: string
+): Promise<[{ sign: string; createdAt: string; text: string }]> {
+  try {
+    const db = database.firestore();
+    const signDataRef = db.collection("signs_ZodiacDirBG").doc(sign);
+    const doc = await signDataRef.get();
+    if (!doc.exists) {
+      console.log("No such document!");
+    } else {
+      return doc.get("data");
+    }
+  } catch (err) {
+    console.log(err);
+  }
+  return [{ sign: sign, createdAt: "", text: "" }];
+}
+
+export async function GetLastElementJSNatFromSignData(
+  sign: string
+): Promise<ISignRecord> {
+  try {
+    const db = database.firestore();
+    const signDataRef = db.collection("signs_ZodiacDirBG").doc(sign);
+    const doc = await signDataRef.get();
+    if (!doc.exists) {
+      console.log("No such document!");
+    } else {
+      console.log(`Document data for SIGN ${sign}: `, doc.data());
+      const fieldton = doc.get("data");
+      const fieldOnlyOne = fieldton[fieldton.length - 1];
+      return {
+        sign: sign,
+        createdAt: fieldOnlyOne["createdAt"],
+        text: fieldOnlyOne["text"],
+      };
+    }
+  } catch (err) {
+    console.log(err);
+  }
+  return {
+    sign: sign,
+    createdAt: "",
+    text: "",
+  };
+}
+
+/*
+// Add today data object to the end of the data array 
+*/
+export async function AppendToArrayTodayElement(
+  arrayData: [SignData],
+  todayElement: SignData
+): Promise<SignData[]> {
+  arrayData.push(todayElement);
+  return arrayData;
+}
+
+export async function isTodayRecordAbsent(
```

Comment:
Instead of doing entire trip to the database - just pass as object the property to this function and then return the boolean. This way you can move this function into helpers folder